crates/sui-proxy/src/main.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use core::time::Duration;
+use std::net::SocketAddr;
+use sui_proxy::admin::{
+    app, create_server_cert_default_allow, create_server_cert_enforce_peer, server,
+};
+use sui_proxy::consumer::UpstreamConsumerBuilder;
+use sui_tls::TlsAcceptor;
+use telemetry_subscribers::TelemetryConfig;
+use tracing::info;
+
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
+#[clap(name = env!("CARGO_BIN_NAME"))]
+#[clap(version = VERSION)]
+struct Args {
+    #[clap(
+        long,
+        short,
+        default_value = "localhost",
+        help = "Specify the tls self-signed cert hostname to use"
+    )]
+    hostname: String,
+    #[clap(long, short, help = "Specify the network name to use for labels")]
+    network: String,
+    #[clap(
+        long,
+        short,
+        default_value = "127.0.0.1:8080",
+        help = "Specify address to listen on"
+    )]
+    listen_address: SocketAddr,
+    #[clap(
+        long,
+        short,
+        default_value = "http://metrics-gw.testnet.sui.io/api/v1/push", // TODO upgrade to https
+        help = "Specify an upstream https url to send to"
+    )]
+    mimir_url: String,
+    #[clap(
+        long,
+        short,
+        default_value = "http://localhost:9000",
+        help = "Specify the remote_write endpoint to use"
+    )]
+    rpc_url: String,
```

Comment:
how is this different from the mimir_url ?

Reply:
or is this a fullnode?

Reply:
the remote_write endpoint is mimir in this case, but could be any remote_write endpoint that implements that protocol.